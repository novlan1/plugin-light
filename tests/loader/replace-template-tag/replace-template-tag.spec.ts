import replaceTmpTag from '../../../src/webpack-loader/replace-template-tag';

jest.mock('loader-utils', () => ({
  getOptions: jest.fn()
    .mockReturnValue({
      replaceTmpTagMap: {
        REPLACE_TAG_SCROLL_VIEW: {
          web: 'div',
          mp: 'scroll-view',
        },
      },
    }),
}));

beforeAll(() => {
  process.env.UNI_PLATFORM = 'mp-weixin';
});


describe('replaceTmpTag', () => {
  const source = `
<template>
  <!-- 赛程树 -->
  <REPLACE_TAG_SCROLL_VIEW
    v-if="col>=1"
    id="scheduleTreeId"
    ref="scheduleTree"
    scroll-y="true"
    scroll-x="true"
    enable-flex="true"
    enhanced="true"
    :class="['tip-match-pk-schedule-list',
            'tip-match-pk-' + teamTotal,
            isLoserTree && isfirstTree ? 'tip-match-lose-tree':'',
            isLoserTree && !isfirstTree ? 'tip-match-lose-line': ''
    ]"
    :style="{ width: \`\${tabWidth}%\`, transform: \`translate3d(\${scheduleScrollNum}rem,0,0)\` }"
    @touchstart="e=>start(e, 't')"
    @touchmove="move"
    @touchend="end"
    @scroll="onScroll"
  >
    <!-- @mouseup="end" -->
  </REPLACE_TAG_SCROLL_VIEW>
</template>
    `;
  it('replaceTmpTag in mp', () => {
    process.env.UNI_PLATFORM = 'mp-weixin';
    expect(replaceTmpTag(source)).toMatchSnapshot();
  });

  it('replaceTmpTag in web', () => {
    process.env.UNI_PLATFORM = 'h5';
    expect(replaceTmpTag(source)).toMatchSnapshot();
  });
});
