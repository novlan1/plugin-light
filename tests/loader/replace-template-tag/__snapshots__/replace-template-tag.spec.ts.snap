// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceTmpTag replaceTmpTag in mp 1`] = `
"
<template>
  <!-- 赛程树 -->
  <scroll-view
    v-if="col>=1"
    id="scheduleTreeId"
    ref="scheduleTree"
    scroll-y="true"
    scroll-x="true"
    enable-flex="true"
    enhanced="true"
    :class="['tip-match-pk-schedule-list',
            'tip-match-pk-' + teamTotal,
            isLoserTree && isfirstTree ? 'tip-match-lose-tree':'',
            isLoserTree && !isfirstTree ? 'tip-match-lose-line': ''
    ]"
    :style="{ width: \`\${tabWidth}%\`, transform: \`translate3d(\${scheduleScrollNum}rem,0,0)\` }"
    @touchstart="e=>start(e, 't')"
    @touchmove="move"
    @touchend="end"
    @scroll="onScroll"
  >
    <!-- @mouseup="end" -->
  </scroll-view>
</template>
    "
`;

exports[`replaceTmpTag replaceTmpTag in web 1`] = `
"
<template>
  <!-- 赛程树 -->
  <div
    v-if="col>=1"
    id="scheduleTreeId"
    ref="scheduleTree"
    scroll-y="true"
    scroll-x="true"
    enable-flex="true"
    enhanced="true"
    :class="['tip-match-pk-schedule-list',
            'tip-match-pk-' + teamTotal,
            isLoserTree && isfirstTree ? 'tip-match-lose-tree':'',
            isLoserTree && !isfirstTree ? 'tip-match-lose-line': ''
    ]"
    :style="{ width: \`\${tabWidth}%\`, transform: \`translate3d(\${scheduleScrollNum}rem,0,0)\` }"
    @touchstart="e=>start(e, 't')"
    @touchmove="move"
    @touchend="end"
    @scroll="onScroll"
  >
    <!-- @mouseup="end" -->
  </div>
</template>
    "
`;
